scripts/qa_crowbarsetup.sh

```diff
@@ -1609,20 +1609,61 @@ function set_proposalvars()
     # Determine if we went through an upgrade
     [ -f /etc/cloudsource ] && export cloudsource=$(</etc/cloudsource)
 
-    wantswift=1
-    [ -z "$want_swift" ] && wantceph=1
-    [[ -n "$wanthyperv" ]] && wantswift= && wantceph= && networkingmode=vlan
+    ### dynamic defaults
+    case "$nodenumber" in
+        0|1)
+            deployswift=
+            deployceph=
+        ;;
+        2)
+            deployswift=1
+            deployceph=
+        ;;
+        *)
+            deployswift=
+            deployceph=1
+        ;;
+    esac
+
+    ### filter (temporarily changing defaults)
+    # F1: hyperV only without swift and ceph
+    [ -n "$wanthyperv" ] && deployswift= && deployceph= && networkingmode=vlan
```

Comment:
Slightly nitpicking, but IMHO this is bad style because it's an unconventional, inconsistent, and semantically overloaded use of `&&`.  `&&` is supposed to be used for conditional expressions, not for sequences of commands.  Furthermore, it can lead to bugs if the code is modified without care to avoid false statements in the pipeline.

So I would prefer this:

    if [ -n "$wanthyperv" ]; then
        deployswift=
        deployceph=
        networkingmode=vlan
    fi

To me this is logically and visually clearer.  If you really want to do all assignments on a single line, you still can:

        deployswift= deployceph= networkingmode=vlan

but it's definitely not as legible, because you're relying on visual parsing of single spaces to understand what gets assigned to what, and the two-dimensional tabular layout is quicker for the brain to parse.  It's also slightly more awkward to edit, because it's harder to navigate between assignments, duplicate an assignment etc.